config/config.go

```diff
@@ -41,6 +44,48 @@ type File struct {
 	Observability *Telemetry `mapstructure:"observability" validate:"required"`
 }
 
+// LoadModulesEndpoint loads the endpoint url and headers for each module.
+// If the endpoint id is not found in the endpoints list, it will use the endpoint id as the url.
+func (f *File) LoadModulesEndpoint() error {
+	for index, module := range f.Node.RSS {
+		endpoint, found := lo.Find(f.Endpoints, func(endpoint Endpoint) bool { return endpoint.ID == module.EndpointID })
+		if !found {
+			endpoint = Endpoint{
+				ID:  module.EndpointID,
+				URL: module.EndpointID,
+			}
+		}
+
+		f.Node.RSS[index].Endpoint = endpoint
+	}
+
+	for index, module := range f.Node.Federated {
+		endpoint, found := lo.Find(f.Endpoints, func(endpoint Endpoint) bool { return endpoint.ID == module.EndpointID })
+		if !found {
+			endpoint = Endpoint{
+				ID:  module.EndpointID,
+				URL: module.EndpointID,
+			}
+		}
+
+		f.Node.Federated[index].Endpoint = endpoint
+	}
+
+	for index, module := range f.Node.Decentralized {
+		endpoint, found := lo.Find(f.Endpoints, func(endpoint Endpoint) bool { return endpoint.ID == module.EndpointID })
+		if !found {
+			endpoint = Endpoint{
+				ID:  module.EndpointID,
+				URL: module.EndpointID,
+			}
+		}
+
+		f.Node.Decentralized[index].Endpoint = endpoint
+	}
```

Comment:
why not using a helper func to avoid repeated logic

```golang
assignEndpoint := func(modules []*Module) {
		for index, module := range modules {
			endpoint, found := lo.Find(f.Endpoints, func(endpoint Endpoint) bool { return endpoint.ID == module.EndpointID })
			if !found {
				endpoint = Endpoint{
					ID:  module.EndpointID,
					URL: module.EndpointID,
				}
			}

			modules[index].Endpoint = endpoint
		}
	}
```